presto-main/src/test/java/com/facebook/presto/execution/resourceGroups/TestResourceGroups.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.resourceGroups;
+
+import com.facebook.presto.execution.MockQueryExecution;
+import com.facebook.presto.execution.resourceGroups.ResourceGroup.RootResourceGroup;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.airlift.units.DataSize;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.execution.QueryState.QUEUED;
+import static com.facebook.presto.execution.QueryState.RUNNING;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static org.testng.Assert.assertEquals;
+
+public class TestResourceGroups
+{
+    @Test(timeOut = 10_000)
+    public void testFairEligibility()
+    {
+        RootResourceGroup root = new RootResourceGroup("root", 1, 4, new DataSize(1, MEGABYTE), MoreExecutors.directExecutor());
+        ResourceGroup group1 = root.addSubGroup("1", 1, 4, new DataSize(1, MEGABYTE));
+        ResourceGroup group2 = root.addSubGroup("2", 1, 4, new DataSize(1, MEGABYTE));
+        MockQueryExecution query1 = new MockQueryExecution(0);
+        group1.add(query1);
+        assertEquals(query1.getState(), RUNNING);
```

Comment:
I feel like `ResourceGroup` is missing methods for monitoring or getting the current state, because I'd like to the tests checking the state of the groups.